client-src/elements/form-field-specs.ts

```diff
@@ -2033,59 +2088,73 @@ function findMaxMilestone(fieldName, stageTypes, getFieldValue) {
 
 // Check that the earlier milestone is before all later milestones.
 // Used with OT start milestone and all shipped milestones.
-function checkEarlierBeforeAllLaterMilestones(fieldPair, getFieldValue) {
+function checkEarlierBeforeAllLaterMilestones(
+  fieldPair: MilestoneRange,
+  getFieldValue: any
+) {
   const {earlier, allLater, warning} = fieldPair;
   const stageTypes =
     // Only shipping, for now.
-    SHIPPED_MILESTONE_FIELDS.has(allLater) ? STAGE_TYPES_SHIPPING : null;
+    allLater && SHIPPED_MILESTONE_FIELDS.has(allLater)
+      ? STAGE_TYPES_SHIPPING
+      : null;
   const earlierValue = getNumericValue(earlier, getFieldValue);
-  const laterValue = findMinMilestone(allLater, stageTypes, getFieldValue);
-  if (
-    earlierValue != null &&
-    laterValue != null &&
-    Number(earlierValue) >= laterValue
-  ) {
-    return warning
-      ? {
-          warning,
-        }
-      : {
-          error:
-            error ||
-            `Earlier milestone #${earlierValue} should be before shipped milestone #${laterValue}.`,
-        };
+  if (stageTypes && allLater) {
+    const laterValue = findMinMilestone(allLater, stageTypes, getFieldValue);
+    if (
+      earlierValue != null &&
+      laterValue != null &&
+      Number(earlierValue) >= laterValue
+    ) {
+      return warning
+        ? {
+            warning,
+          }
+        : {
+            error: `Earlier milestone #${earlierValue} should be before shipped milestone #${laterValue}.`,
+          };
+    }
   }
 }
 
 // Check that all earlier milestones before a later milestone.
 // Used with all OT start milestones and a shipped milestone.
-function checkAllEarlierBeforeLaterMilestone(fieldPair, getFieldValue) {
-  const {allEarlier, later, warning} = fieldPair;
+function checkAllEarlierBeforeLaterMilestone(
+  fieldPair: MilestoneRange,
+  getFieldValue: any
+) {
+  const {allEarlier, later, warning, error} = fieldPair;
   const stageTypes =
+    allEarlier &&
     // Only origin trials or dev trials, for now.
     OT_MILESTONE_START_FIELDS.has(allEarlier)
       ? STAGE_TYPES_ORIGIN_TRIAL
-      : DT_MILESTONE_FIELDS.has(allEarlier)
+      : allEarlier && DT_MILESTONE_FIELDS.has(allEarlier)
         ? STAGE_TYPES_DEV_TRIAL
         : null;
   // console.info(`stageTypes: ${stageTypes}`);
```

Comment:
Not related to this PR, but I believe these commented out console info statements can be removed.